xds/internal/balancer/ringhash/util.go

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package ringhash contains the functionality to support Ring Hash in grpc.
+package ringhash
+
+import "context"
+
+type clusterKey struct{}
+
+func getRequestHash(ctx context.Context) uint64 {
+	requestHash, _ := ctx.Value(clusterKey{}).(uint64)
+	return requestHash
+}
+
+// GetRequestHashForTesting returns the request hash in the context; to be used
+// for testing only.
+func GetRequestHashForTesting(ctx context.Context) uint64 {
```

Comment:
Hmm, this function is never called? Delete it?

Reply:
Actually, add a test case to `TestXDSResolverGoodServiceUpdate`, and call this to get the hash from the context (similar to how it calls `clustermanager.GetPickedClusterForTesting`)

Reply:
Added my own test. That test was testing a certain functionality, and was a little too jumbled to add extra functionality to it. I wrote my own that specifically validated request hash generated in SelectConfig.